opossum-tool/OpossumUI
# Improve locate popup handling

### Summary of changes

- Close popup when pressing apply and signals are located or no filter is set
- Inform user when apply is pressed and no signals are located
- Only change state when clicking on "apply" (not when clicking "clear")
- Test that the message is shown

### Context and reason for change

We are introducing a feature to "locate" resources that have or contain attributions with criteria that can be set in a new popup.

### How can the changes be tested

Open OpossumUI, select "Locate Signals" in the edit context. In the popup, set the filters to search for a signal that does not exist, press "apply" and observe that the message is shown. Then, search for an existing signal, press "apply" and verify that the popup closes.

Note: Please review the [guidelines for contributing](https://github.com/opossum-tool/OpossumUI/blob/main/CONTRIBUTING.md) to this repository.
